src/main/java/com/epam/bookstoreservice/service/impl/BookstoreServiceImpl.java

```diff
@@ -30,144 +34,173 @@ public class BookstoreServiceImpl implements BookstoreService {
 
     private final BookDao bookDao;
 
+    private static final String SUCCESSFUL_MESSAGE = "successful";
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
 
     @Override
-    public BookResponseDto addNewBook(BookRequestDto book) {
+    public Result<BookResponseDTO> addNewBook(BookRequestDTO book) {
         BookEntity bookEntity = bookRequestDtoToBookEntity.requestDtoToEntity(book);
         BookEntity newBookEntity = bookDao.save(bookEntity);
-        return bookRequestDtoToBookEntity.entityToResponseDto(newBookEntity);
+        return Result.success(SUCCESSFUL_MESSAGE, bookRequestDtoToBookEntity.entityToResponseDto(newBookEntity));
     }
 
     @Override
-    public BookResponseDto addBook(BookRequestDto book) {
+    public Result<BookResponseDTO> addBook(BookRequestDTO book) {
         BookEntity bookEntity = bookDao.findByTitle(book.getTitle());
 
         Integer totalCount = book.getTotalCount() + bookEntity.getTotalCount();
         bookEntity.setTotalCount(totalCount);
         BookEntity result = bookDao.save(bookEntity);
 
-        return bookRequestDtoToBookEntity.entityToResponseDto(result);
+        return Result.success(SUCCESSFUL_MESSAGE, bookRequestDtoToBookEntity.entityToResponseDto(result));
     }
 
     @Override
-    public BookResponseDto getBookById(Integer id) {
+    public Result<BookResponseDTO> getBookById(Integer id) {
         Optional<BookEntity> bookById = bookDao.findById(id);
+
+        if (!bookById.isPresent()) {
```

Comment:
You can use bookById.orElseThrow() to simplify your code.
Try to use a more specific name than "bookById".